kernel/aster-nix/src/fs/file_table.rs

```diff
@@ -232,6 +238,39 @@ pub struct FileTableEntry {
     file: Arc<dyn FileLike>,
     flags: AtomicU8,
     subject: Subject<FdEvents>,
+    owner: RwLock<Option<Owner>>,
+}
+
+type Owner = (Pid, Arc<dyn Observer<IoEvents>>);
+
+struct OwnerObserver {
+    file: Arc<dyn FileLike>,
+    owner: Pid,
```

Comment:
I think it is more reliable to keep a weak reference to the owner process. The owner process may exit and its pid will be re-assigned to other processes. In this case, the signal may be sent to the wrong process.

```rust
impl Observer<IoEvents> for OwnerObserver {
    fn on_events(&self, events: &IoEvents) {
        if self.file.status_flags().contains(StatusFlags::O_ASYNC)
            && let Some(process) = process_table::get_process(self.owner)
```

Furthermore, I/O events may occur quite frequently. So the `on_events` method should be implemented as efficient as possible. `process_table::get_process` will have to lock the entire process table. So it is relatively costly. Instead, if the `OwnerObserver` keeps a weak reference to the owner process, then the `on_events` method only needs to do `Weak::upgrade`, which is much cheaper than `on_events`.